builder/base_image.go

```diff
@@ -25,53 +26,75 @@ func NewBaseImageBuilder(baseImage string) *BaseImageBuilder {
 	}
 }
 
-// CreateDockerfileDir generates a Dockerfile and assets in the targetDir
-func (b *BaseImageBuilder) CreateDockerfileDir(targetDir, configginTarballPath string) error {
-	if err := os.MkdirAll(targetDir, 0755); err != nil {
-		return err
+// PopulateDockerArchive returns a function that will populate the docker tar archive
+func (b *BaseImageBuilder) PopulateDockerArchive(configginTarballPath string) func(*tar.Writer) error {
+	return func(tarWriter *tar.Writer) error {
+		// Generate dockerfile
+		dockerfileContents, err := b.generateDockerfile()
+		if err != nil {
+			return err
+		}
+		err = util.WriteToTarStream(tarWriter, dockerfileContents, tar.Header{
+			Name: "Dockerfile",
+		})
+		if err != nil {
+			return err
+		}
+
+		// Add monitrc
+		monitrcContents, err := dockerfiles.Asset("monitrc.erb")
+		if err != nil {
+			return err
+		}
+		err = util.WriteToTarStream(tarWriter, monitrcContents, tar.Header{
+			Name: "monitrc.erb",
+			Mode: 0600,
+		})
+		if err != nil {
+			return err
+		}
+
+		// Add rsyslog_conf
+		for _, assetName := range dockerfiles.AssetNames() {
+			if !strings.HasPrefix(assetName, "rsyslog_conf/") {
+				continue
+			}
+			assetContents, err := dockerfiles.Asset(assetName)
+			if err != nil {
+				return err
+			}
+			err = util.WriteToTarStream(tarWriter, assetContents, tar.Header{
+				Name: assetName,
+			})
+			if err != nil {
+				return err
+			}
+		}
+
+		// Add configgin
+		configginGzip, err := ioutil.ReadFile(configginTarballPath)
```

Comment:
Yeah, this _will_ go away (that's the actual point of this task), so I'm just being lazy here.  Though you're right, I should just have had a reader. Whatever :p